internal/attestation/azure/snp/maa.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package snp
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+
+	"github.com/edgelesssys/go-azguestattestation/maa"
+)
+
+type maaClient struct {
+	url    string // URL of the MAA service. Potentially configured at runtime.
+	client *http.Client
+}
+
+func newMAAClient(url string) *maaClient {
+	return &maaClient{
+		url:    url,
+		client: http.DefaultClient,
+	}
+}
+
+func (m *maaClient) createToken(ctx context.Context, data []byte) (string, error) {
+	return maa.Attest(ctx, data, m.url, m.client)
```

Comment:
We could use maa.NewParameters etc. here if we wanted to remove some code from the snp package that does the same.

Reply:
Requires some [changes in the MAA repo](https://github.com/edgelesssys/go-azguestattestation/pull/2), or we can keep the current implementation (which just copies some unexported code)